server/forwarding.go

```diff
@@ -49,8 +49,12 @@ func (s *server) ServeDNSForward(w dns.ResponseWriter, req *dns.Msg) *dns.Msg {
 		err error
 		try int
 	)
-	// Use request Id for "random" nameserver selection.
-	nsid := int(req.Id) % len(s.config.Nameservers)
+
+	nsid := 0
+	if s.config.RoundRobin {
```

Comment:
you're overloading the roundrobin flag here, so a new flag would be better.

Also why do you give skydns multiple dns servers and you only want one to be used?